streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKTableLeftJoinTest.java

```diff
@@ -45,7 +47,13 @@
 import java.util.Random;
 import java.util.Set;
 
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
```

Comment:
This is not decided on the project level, but I personally find it confusing that a few classes use hamcrest matchers and most others use JUnit5 basic assertions. I'd not introduce new classes using hamcrest. See also https://issues.apache.org/jira/browse/KAFKA-16929. But feel free to ignore since this more of a project-level concern to move away from hamcrest.

Reply:
Happy to make this change to avoid downstream work in case we do decide to not use hamcrest.